Source/Particles/ParticleBoundaryBuffer.cpp

```diff
@@ -0,0 +1,223 @@
+/* Copyright 2021 Andrew Myers
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+
+#include "EmbeddedBoundary/DistanceToEB.H"
+#include "Particles/ParticleBoundaryBuffer.H"
+#include "Particles/MultiParticleContainer.H"
+
+#include <AMReX_Geometry.H>
+#include <AMReX_ParmParse.H>
+
+ParticleBoundaryBuffer::ParticleBoundaryBuffer ()
+{
+    m_particle_containers.resize(numBoundaries());
+
+    for (int i = 0; i < numBoundaries(); ++i)
+    {
+        m_particle_containers[i].resize(numSpecies());
+    }
+
+    m_do_boundary_buffer.resize(numBoundaries());
+    for (int i = 0; i < numBoundaries(); ++i)
+    {
+        m_do_boundary_buffer[i].resize(numSpecies(), 0);
+    }
+
+    for (int i = 0; i < numBoundaries(); ++i)
+    {
+        for (int ispecies = 0; ispecies < numSpecies(); ++ispecies)
+        {
+            amrex::ParmParse pp_species(getSpeciesNames()[ispecies]);
+            pp_species.query("scrape_xlo", m_do_boundary_buffer[0][ispecies]);
```

Comment:
Instead of `scrape_xlo`, could this be more descriptive, perhaps `save_particles_at_xlo`?

Reply:
Yes, done.